pages/api/augment-request-gql.ts

```diff
@@ -0,0 +1,408 @@
+  switch (oracleType) {
+    case "OptimisticOracle":
+      return "Optimistic";
+    case "OptimisticOracleV2":
+      return "OptimisticV2";
+    case "SkinnyOptimisticOracle":
+      return "Skinny";
+    case "OptimisticOracleV3":
+      return "OptimisticV3";
+    default:
+      throw new Error("Unable to cast oracle name for OO UI: " + oracleType);
+  }
+}
+
+async function voteQuery({
+  time,
+  identifier,
+  ancillaryData,
+}: Omit<OracleQueryParams, "chainId">) {
+  const id = makeUniqueKeyForVote(identifier, time, ancillaryData);
+  const query = gql`
+    query voteQuery($id: ID!) {
+      priceRequest(id: $id) {
+        requestTransaction
+      }
+    }
+  `;
+  type GqlRequest = {
+    requestTransaction: string;
+  };
+  type GqlResponse = {
+    priceRequest: GqlRequest | undefined | null;
+  };
+  try {
+    const data: GqlResponse = await request(VoteSubgraphURL, query, { id });
+    assert(data.priceRequest, "request not found");
+    return data?.priceRequest;
+  } catch (error) {
+    if (debug) console.error("vote fetch error:", error);
+    throw error;
+  }
+}
+async function ooSkinnyQuery({
+  time,
+  identifier,
+  chainId,
+}: OracleQueryParams): Promise<ResponseBody> {
+  const subgraph = getSubgraphConfig("Skinny Optimistic Oracle", chainId);
+  const query = gql`
+    query skinnyQuery($proposalTimestamp: Int!) {
+      optimisticPriceRequests(
+        where: { proposalTimestamp: $proposalTimestamp }
+      ) {
+        id
+        requestHash
+        requestLogIndex
+      }
+    }
+  `;
+  type GqlRequest = {
+    id: string;
+    requestHash: string;
+    requestLogIndex: string;
+  };
+  type GqlResponse = {
+    optimisticPriceRequests: GqlRequest[];
+  };
+
+  try {
+    const data: GqlResponse = await request(subgraph.url, query, {
+      proposalTimestamp: time,
+    });
+    assert(data.optimisticPriceRequests.length > 0, "request not found");
+    const [optimisticPriceRequest] = data.optimisticPriceRequests;
+    const requestHash = optimisticPriceRequest?.requestHash;
+    const requestLogIndex = optimisticPriceRequest?.requestLogIndex;
+    return {
+      time,
+      uniqueKey: optimisticPriceRequest.id,
+      identifier,
+      ooRequestUrl: constructOoUiLink(
+        requestHash,
+        chainId,
+        "SkinnyOptimisticOracle",
+        requestLogIndex
+      ),
+      originatingChainTxHash: requestHash,
+      originatingChainId: chainId,
+      originatingOracleType: "SkinnyOptimisticOracle",
+    };
+  } catch (error) {
+    if (debug) console.error("oo skinny error:", error);
+    throw error;
+  }
+}
+async function oov3Query({
+  time,
+  identifier,
+  chainId,
+}: OracleQueryParams): Promise<ResponseBody> {
+  const subgraph = getSubgraphConfig("Optimistic Oracle V3", chainId);
+  const query = gql`
+    query oov3Query($assertionTimestamp: Int!) {
+      assertions(where: { assertionTimestamp: $assertionTimestamp }) {
```

Comment:
`assertionTimestamp` doesn't uniquely identify an assertion
We should use the `ancillaryData` from the VotingV2 price request, parse it to extract the `assertionId` and query the subgraph by `assertionId`

See how the ancillaryData is calculated in the OOv3 [here](https://github.com/UMAprotocol/protocol/blob/454cd8aed43094c94916d6225e4c52403d8bdf19/packages/core/contracts/optimistic-oracle-v3/implementation/OptimisticOracleV3.sol#L394)

Reply:
ive updated this to be more specific